phpBB/phpbb/language/language.php

```diff
@@ -0,0 +1,590 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\language;
+
+use phpbb\language\exception\invalid_plural_rule_exception;
+
+/**
+ * Wrapper class for loading translations
+ */
+class language
+{
+	/**
+	 * @var array	List of common language files
+	 */
+	protected $common_language_files;
+
+	/**
+	 * @var bool
+	 */
+	protected $common_language_files_loaded;
+
+	/**
+	 * @var string	ISO code of the default board language
+	 */
+	protected $default_language;
+
+	/**
+	 * @var array
+	 */
+	protected $help;
+
+	/**
+	 * @var array	Array of language variables
+	 */
+	protected $lang;
+
+	/**
+	 * @var string|null	Custom path to language files
+	 */
+	protected $language_root_path;
+
+	/**
+	 * @var array	Loaded language sets
+	 */
+	protected $loaded_language_sets;
+
+	/**
+	 * @var 	Language loader factory
+	 */
+	protected $loader_factory;
+
+	/**
+	 * @var string	ISO code of the User's language
+	 */
+	protected $user_language;
+
+	/**
+	 * Constructor
+	 */
+	public function __construct($container)
+	{
+		// @todo: replace this with loader factory
+		$this->loader_factory = $container;
+
+		// Set up default information
+		$this->user_language		= false;
+		$this->default_language		= false;
+		$this->language_root_path	= null;
+		$this->lang					= array();
+		$this->loaded_language_sets	= array(
+			'core'	=> array(),
+			'ext'	=> array(),
+		);
+
+		// Common language files
+		$this->common_language_files = array(
+			'common',
```

Comment:
I think it could be nice to be able to configure it through an optional parameter in the DI